### PR TITLE
Improve assertions

### PR DESCRIPTION
**Pull request by fernandodev**
For this branch I pick all assertion and message changes from PR #16 

So after merging this, PR #16 shouldn't be merged. I'm working to split #16 into 2 other PRs.

### CODE REVIEW

File: XWebView/XWVHttpConnection.m [hunk 1]
```diff
@@ -341,7 +341,8 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode {
     NSString *name;
 
     int class = (int)response.statusCode / 100 - 1;
-    assert(class >= 0 && class < 5);
+    NSCAssert(class >= 0 && class < 5, @"<XWV> status code must be in the range [0, 500)");
```
Comment (xwv) [inline XWebView/XWVHttpConnection.m]: According to [RFC7231 section 6](https://tools.ietf.org/html/rfc7231#section-6), status code should be in range [100, 599].

File: XWebView/XWVChannel.swift [hunk 2]
```diff
@@ -54,8 +56,10 @@ public class XWVChannel : NSObject, WKScriptMessageHandler {
     }
 
     public func bindPlugin(object: AnyObject, toNamespace namespace: String) -> XWVScriptObject? {
-        assert(webView != nil && typeInfo == nil)
-        webView!.configuration.userContentController.addScriptMessageHandler(self, name: name)
+        assert(typeInfo == nil, "<XWV> This channel already have an bound object")
```
Comment (xwv) [inline XWebView/XWVChannel.swift]: This channel already **has a** bound object

**Comment by xwv** [comment]
LGTM with 2 nits, thanks.

File: XWebView/XWVHttpConnection.m [hunk 3]
```diff
@@ -105,7 +105,7 @@ - (NSURL *)rootURL {
     NSURL *root;
     if (_delegate && [_delegate respondsToSelector:@selector(documentRoot)]) {
         root = [NSURL fileURLWithPath:_delegate.documentRoot isDirectory:YES];
-        assert(root);
+        NSCAssert(root != nil, @"<XWV> you must set a valid documentRoot");
```
Comment (xwv) [inline XWebView/XWVHttpConnection.m]: According to [Foundation Functions Reference](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Miscellaneous/Foundation_Functions/#//apple_ref/c/macro/NSAssert), NSCAssert should be used only in C function. In ObjC method, NSAssert is the proper macro.

**Comment by fernandodev** [comment]
@xwv done :+1:

File: XWebView/XWVHttpConnection.m (hunk 3 above)
Comment (fernandodev) [inline XWebView/XWVHttpConnection.m]: I really don't know how the proper way to use `NSAssert` and its variants. 
I get `use of undeclared identifier _cmd` when I try to use it.
I've looked for this error and generally people use `NSCAssert` but I really agree with you after reading documentation with more attention.
So any kind of idea?

Comment (xwv) [inline XWebView/XWVHttpConnection.m]: Just replacing NSCAssert with NSAssert works on my machine, no error or warning.

```
NSAssert(root != nil, @"<XWV> you must set a valid documentRoot");
```

**Pull request merged**